### ISSUE TITLE
Cannot parse cisco smtp community with plus sign in it

### ISSUE THREAD

**Issue by virtuald** [batfish/batfish#360]
It seems that you can have special characters in an smtp community, such as:

```
snmp-server community dummy+dummy RO 1
snmp-server community dummy@dummy RW 2
```

I've played with the grammar for quite a bit, but I'm afraid my ANTLR-fu is not strong enough to fix this. It seems that because PLUS is a token already, it won't be included as part of a variable (as ss_community requires at https://github.com/batfish/batfish/blob/master/projects/batfish/src/main/antlr4/org/batfish/grammar/cisco/Cisco_snmp.g4#L38). My only guess is that maybe you need a special mode to parse this variable?

You can duplicate this bug by adding the above community lines around https://github.com/batfish/batfish/blob/master/test_rigs/unit-tests/configs/cisco_snmp#L6

CC @jkhourybbn

**Comment by arifogel** [comment]
The solution is to make a different variable_xxx rule (see Cisco_common.g4) that aggregates the regular variable rule with `+` and `@`. What's tricky is making sure it stops before it consumes too much. I got this one.

**Comment by virtuald** [comment]
@arifogel I pushed a fix for this in #384 yesterday -- though, if someone inserted some whitespace between the `+` or `@` symbols then it would interpret the name incorrectly.
